plugins/base/src/main/kotlin/transformers/documentables/DocumentableVisibilityFilter.kt

```diff
@@ -0,0 +1,299 @@
+                            dri,
+                            documentation.filtered(filteredPlatforms),
+                            sources.filtered(filteredPlatforms),
+                            functions,
+                            properties,
+                            classlikes,
+                            visibility.filtered(filteredPlatforms),
+                            companion,
+                            constructors,
+                            filteredPlatforms,
+                            extra
+                        )
+                        this is Enum -> Enum(
+                            dri,
+                            name,
+                            enumEntries,
+                            documentation.filtered(filteredPlatforms),
+                            sources.filtered(filteredPlatforms),
+                            functions,
+                            properties,
+                            classlikes,
+                            visibility.filtered(filteredPlatforms),
+                            companion,
+                            constructors,
+                            supertypes.filtered(filteredPlatforms),
+                            filteredPlatforms,
+                            extra
+                        )
+                        this is Interface -> Interface(
+                            dri,
+                            name,
+                            documentation.filtered(filteredPlatforms),
+                            sources.filtered(filteredPlatforms),
+                            functions,
+                            properties,
+                            classlikes,
+                            visibility.filtered(filteredPlatforms),
+                            companion,
+                            generics,
+                            supertypes.filtered(filteredPlatforms),
+                            filteredPlatforms,
+                            extra
+                        )
+                        this is Object -> Object(
+                            name,
+                            dri,
+                            documentation.filtered(filteredPlatforms),
+                            sources.filtered(filteredPlatforms),
+                            functions,
+                            properties,
+                            classlikes,
+                            visibility,
+                            supertypes.filtered(filteredPlatforms),
+                            filteredPlatforms,
+                            extra
+                        )
+                        else -> null
+                    }
+                }
+            }
+        }
+        return Pair(classlikesListChanged, filteredClasslikes)
+    }
+
+    private val allowedVisibilities = arrayOf(
+        JavaVisibility.Public,
+        JavaVisibility.Default,
+        JavaVisibility.Protected,
+        KotlinVisibility.Protected,
```

Comment:
I'd personally filter out `Protected` members by default as they are not a part of the API most of the time. 
This list should be configurable by user though or at least we should have some options to choose from